### PR TITLE
Feature/add role and rolebinding

### PR DESCRIPTION
**Pull request by chapipo**
Add template to `lib-k8s-as-helm` to generate some part of [Cluster]Role and [Cluster]Rolebinding.

Add 4 charts for:
- Role
- Rolebinding
- ClusterRole
- ClusterRolebinding

### CODE REVIEW

File: charts/clusterrole/README.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+The following table lists the configurable parameters of the clusterrole chart and their default values.
+
+Parameter | Description | Default
+--- | --- | ---
+`nameOverride` | override name of the chart component | .Release.Name
+`apiVersion` | api version of k8s object | `"v1"`
+`annotations` | annotations in yaml map format to be added to the object | `null`
+`labels` | labels to add to ClusterRole object | `null`
+`rules` | list of policy rules for this ClusterRole | `[]`
+rules[].verbs` | list of verbs that apply to all the ResourceKinds and AttributeRestrictions in this rule | `null`
```
Comment (ameijer) [inline charts/clusterrole/README.md:37]:
```suggestion
`rules[].verbs` | list of verbs that apply to all the ResourceKinds and AttributeRestrictions in this rule | `null`
```

File: charts/clusterrole/README.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+```
+
+## Configuration
+
+The following table lists the configurable parameters of the clusterrole chart and their default values.
+
+Parameter | Description | Default
+--- | --- | ---
+`nameOverride` | override name of the chart component | .Release.Name
+`apiVersion` | api version of k8s object | `"v1"`
```
Comment (ameijer) [inline charts/clusterrole/README.md:33]:
```suggestion
`apiVersion` | api version of k8s object | `"rbac.authorization.k8s.io/v1"`
```

File: charts/clusterrolebinding/Chart.yaml [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+icon: https://ameijer.github.io/k8s-as-helm/icon.png
+dependencies:
+- name: lib-k8s-as-helm
+  version: 1.2.0
+  repository: file://../lib-k8s-as-helm
+maintainers:
+  - name: ameijer
+    url: https://github.com/ameijer
+keywords:
+- rolebinding
```
Comment (ameijer) [inline charts/clusterrolebinding/Chart.yaml:16]:
```suggestion
- clusterrolebinding
```

File: charts/rolebinding/README.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+```
+
+## Configuration
+
+The following table lists the configurable parameters of the rolebinding chart and their default values.
+
+Parameter | Description | Default
+--- | --- | ---
+`nameOverride` | override name of the chart component | .Release.Name
+`apiVersion` | api version of k8s object | `"v1"`
```
Comment (ameijer) [inline charts/rolebinding/README.md:33]:
```suggestion
`apiVersion` | api version of k8s object | `"rbac.authorization.k8s.io/v1"`
```

File: charts/role/README.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+```
+
+## Configuration
+
+The following table lists the configurable parameters of the role chart and their default values.
+
+Parameter | Description | Default
+--- | --- | ---
+`nameOverride` | override name of the chart component | .Release.Name
+`apiVersion` | api version of k8s object | `"v1"`
```
Comment (ameijer) [inline charts/role/README.md:33]:
```suggestion
`apiVersion` | api version of k8s object | `"rbac.authorization.k8s.io/v1"`
```

File: charts/clusterrolebinding/README.md [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+```
+
+## Configuration
+
+The following table lists the configurable parameters of the clusterrolebinding chart and their default values.
+
+Parameter | Description | Default
+--- | --- | ---
+`nameOverride` | override name of the chart component | .Release.Name
+`apiVersion` | api version of k8s object | `"v1"`
```
Comment (ameijer) [inline charts/clusterrolebinding/README.md:33]:
```suggestion
`apiVersion` | api version of k8s object | `"rbac.authorization.k8s.io/v1"`
```

File: charts/clusterrolebinding/README.md [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+`annotations` | annotations in yaml map format to be added to the object | `null`
+`labels` | labels to add to ClusterRoleBinding object | `null`
+`roleRef` | object referencing a Role in the current namespace or a ClusterRole | `{}`
+`roleRef.apiGroup` | apiGroup of the role referenced | `null`
+`roleRef.kind` | kind of the role referenced | `null`
+`roleRef.name` | name of the role referenced | `null`
+`subjects` | yaml list of subject to apply the role | `{}`
```
Comment (ameijer) [inline charts/clusterrolebinding/README.md:40]:
```suggestion
`subjects` | yaml list of subject to apply the role | `[]`
```

File: charts/clusterrolebinding/values.yaml [hunk 8]
```diff
@@ -0,0 +1 @@
+roleRef: {}
```
Comment (ameijer) [inline charts/clusterrolebinding/values.yaml:1]:
```suggestion
roleRef: {}
subjects: []
```

File: charts/clusterrolebinding/README.md [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+
+The following table lists the configurable parameters of the clusterrolebinding chart and their default values.
+
+Parameter | Description | Default
+--- | --- | ---
+`nameOverride` | override name of the chart component | .Release.Name
+`apiVersion` | api version of k8s object | `"v1"`
+`annotations` | annotations in yaml map format to be added to the object | `null`
+`labels` | labels to add to ClusterRoleBinding object | `null`
+`roleRef` | object referencing a Role in the current namespace or a ClusterRole | `{}`
```
Comment (ameijer) [inline charts/clusterrolebinding/README.md:36]: can cluster role bindings refer to roles? have never tried that

File: charts/rolebinding/README.md [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+`annotations` | annotations in yaml map format to be added to the object | `null`
+`labels` | labels to add to Secret object | `null`
+`roleRef` | object referencing a Role in the current namespace or a ClusterRole | `null`
+`roleRef.apiGroup` | apiGroup of the role referenced | `null`
+`roleRef.kind` | kind of the role referenced | `null`
+`roleRef.name` | name of the role referenced | `null`
+`subjects` | yaml list of subject to apply the role | `null`
```
Comment (ameijer) [inline charts/rolebinding/README.md:40]: empty array as a default value here for consistency with clusterrolebinding?

File: charts/rolebinding/README.md [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+
+The following table lists the configurable parameters of the rolebinding chart and their default values.
+
+Parameter | Description | Default
+--- | --- | ---
+`nameOverride` | override name of the chart component | .Release.Name
+`apiVersion` | api version of k8s object | `"v1"`
+`annotations` | annotations in yaml map format to be added to the object | `null`
+`labels` | labels to add to Secret object | `null`
+`roleRef` | object referencing a Role in the current namespace or a ClusterRole | `null`
```
Comment (ameijer) [inline charts/rolebinding/README.md:36]: empty object here as default for consistency with cluster role binding?

**Review by ameijer — CHANGES_REQUESTED** [review]
this looks great! This is super close. Library chart was definitely a good approach. Just found a few things to look at

File: charts/clusterrolebinding/README.md (hunk 9 above)
Comment (chapipo) [reply charts/clusterrolebinding/README.md]: Role bindings can refer to cluster role but not the inverse.

**Comment by chapipo** [comment]
I just update the branch to solve all change requested.

**Review by ameijer — APPROVED** [review]
lgtm

**Pull request merged**